backend/Auth/ResetPassword.ts

```diff
@@ -115,6 +119,63 @@ router.get<Params, ResponseBody, QueryParams>(
   }
 );
 
+router.put(
+  "/api/passwords",
+  body("password").isStrongPassword(),
+  body("token").exists(),
+  async (req, res) => {
+    const { password, token } = req.body;
+
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      if (token) {
+        return invalidRequest(
+          res,
+          "Password must be at least 8 characters long, contain one number, one uppercase letter, and one special character."
+        );
+      } else {
+        return invalidRequest(
+          res,
+          "Request must include token. Please reset your password again."
+        );
+      }
+    }
+
+    if (!tokenIsValid(token, jwtSecret)) {
+      return invalidRequest(res, "Invalid token");
+    }
+
+    const rows = await JWTModel.findAll({
+      where: {
+        token: token,
+      },
+    });
+
+    if (rows.length >= 1) {
+      const userId = rows[0].userId;
+      const hash = await bcrypt.hash(password, 5);
+
+      const toUpdate = {
+        password: hash,
+      };
+
+      const [numUpdated] = await UserModel.update(toUpdate, {
+        where: {
+          id: userId,
+        },
+      });
+
+      if (numUpdated === 0) {
+        notFound(res, `Unable to update password`);
+      } else {
+        successNoContent(res);
```

Comment:
Minor, but you could return here to make sure that if code is added later on to this function that we don't try to do anything after having sent the HTTP response:

```suggestion
        return successNoContent(res);
```

Same goes for line 169